src/zenml/integrations/huggingface/model_registries/huggingface_model_registry.py

```diff
@@ -0,0 +1,668 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Implementation of the MLflow model registry for ZenML."""
+
+
+import os
+from contextlib import contextmanager
+from datetime import datetime
+from pathlib import Path
+from typing import Any, ClassVar, Dict, List, Optional, Tuple, cast
+from uuid import uuid4
+
+from huggingface_hub import HfApi, ModelCard, ModelCardData
+
+from zenml.integrations.huggingface.flavors.huggingface_model_registry_flavor import (
+    HuggingfaceModelRegistryConfig,
+)
+from zenml.logger import get_logger
+from zenml.model_registries.base_model_registry import (
+    BaseModelRegistry,
+    ModelRegistryModelMetadata,
+    ModelVersion,
+    ModelVersionStage,
+    RegisteredModel,
+)
+
+logger = get_logger(__name__)
+
+
+class HuggingfaceRegisteredModel(RegisteredModel):
+    """Huggingface registered model class."""
+
+    repo_id: str
+
+
+class HuggingfaceModelRegistry(BaseModelRegistry):
+    """Register models using Huggingface."""
+
+    MODELCARD_TEMPLATE_PATH: ClassVar[str] = f"{uuid4()}.md"
+    REPO_TYPE: ClassVar[str] = "model"
+
+    @property
+    def config(self) -> HuggingfaceModelRegistryConfig:
+        """Returns the `HuggingfaceModelRegistryConfig` config.
+
+        Returns:
+            The configuration.
+        """
+        return cast(HuggingfaceModelRegistryConfig, self._config)
+
+    @property
+    def api(self) -> HfApi:
+        return HfApi(token=self.config.token)
+
+    def _model_exists(self, name: str):
+        if not self.api.repo_exists(
+            repo_id=self._repo_id(name), repo_type=self.REPO_TYPE
+        ):
+            raise KeyError(
+                f"Model with name `{self._repo_id(name)}` doesn't exist."
+            )
+
+    @contextmanager
+    def model_card_template(self):
+        template_text = """
+---
+{{ card_data }}
+---
+# {{ model_name }}
+
+## Description
+{{ model_description }}
+
+## Metadata
+{% for key,value in metadata.items() %}
+{{ key }}:  {{ value }}
+{% endfor %}
+
+_This model was created from **ZenML**_
+""".strip()
+        Path(self.MODELCARD_TEMPLATE_PATH).write_text(template_text)
+        yield
+        os.remove(self.MODELCARD_TEMPLATE_PATH)
+
+    def _repo_id(self, name: str) -> str:
+        return f"{self.config.user}/{name}"
```

Comment:
One thing that is somewhat common with the HF hub is that people will have their personal repository / account and then their organisation account and quite often switch between them or sometimes push to one vs the other.

Given that fact, someone following that pattern would basically need to have two registered model registries, right, and either update their stack to change the username or just switch stacks, right? I'm wondering if there's a nicer way to allow for this? (100% not a core concern for this PR, however, just wanted to mention it...)